### PR TITLE
FuseERC4626

### PR DESCRIPTION
**Pull request by eswak**
Add an ERC4626-compliant wrapper around Compound's cTokens

### CODE REVIEW

File: src/vaults/compound/FuseERC4626.sol [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    /// @notice Total amount of the underlying asset that
+    /// is "managed" by Vault.
+    function totalAssets() public view override returns(uint256) {
+        // Use libfuse to determine an accurate view exchange rate.
+        // this use Fuse cToken functions that do not exist in Compound implementations:
+        // - cToken.totalAdminFees()
+        // - cToken.totalFuseFees()
+        // - cToken.adminFeeMantissa()
+        return cToken.viewUnderlyingBalanceOf(address(this));
```
Comment (Joeysantoro) [inline src/vaults/compound/FuseERC4626.sol]: One idea t11s had was to use the `implementation` address of the cToken and revert if it is unexpected so the logic can't change underneath us. I'm not sure if its worth the gas or extra admin overhead but something to consider

**Comment by eswak** [comment]
Here's a forge integration test file to try integration with feirari: [4626-fFEI-8.sol.txt](https://github.com/fei-protocol/ERC4626/files/8199415/4626-fFEI-8.sol.txt)

**Comment by transmissions11** [comment]
Should technically override the max functions to return the max, or else it's non compliant  

If there's not enough cash for withdrawals or there's a deposit limit

Idk how to handle pauses, will ask the other authors

**Comment by Joeysantoro** [comment]
LGTM, lets get one more set of eyes

**Comment by transmissions11** [comment]
also need to return 0 from the max functions if redemption or minting is paused respectively

**Comment by Joeysantoro** [comment]
> also need to return 0 from the max functions if redemption or minting is paused respectively

They do when minting is paused, redemptions can't be paused iiuc

**Review by xklob — COMMENTED** [review]
LGTM

**Pull request merged**